collect_app/src/main/java/org/odk/collect/android/tasks/sms/SmsUtils.java

```diff
@@ -0,0 +1,160 @@
+
+    private SmsUtils() {
+
+    }
+
+    /***
+     * Create the intent that's passed to SMS Manager so that the Sent SMS
+     * BroadcastReceiver can be triggered.
+     * @param context necessary to create pending intent.
+     * @param instanceId identifies the instance of the form being targeted.
+     * @param messageId identifies the specific messages this intent applies to.
+     * @return PendingIntent
+     */
+
+    public static PendingIntent getSentPendingIntent(Context context, String instanceId, int messageId) {
+        Intent sendIntent = new Intent(SMS_SEND_ACTION);
+        sendIntent.putExtra(SMS_INSTANCE_ID, instanceId);
+        sendIntent.putExtra(SMS_MESSAGE_ID, messageId);
+
+        return PendingIntent.getBroadcast(context, messageId, sendIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+    }
+
+
+    /***
+     * Create the intent that's passed to SMS Manager so that the Delivery SMS
+     * BroadcastReceiver can be triggered.
+     * @param context necessary to create pending intent.
+     * @param instanceId identifies the instance of the form being targeted.
+     * @param messageId identifies the specific messages this intent applies to.
+     * @return PendingIntent
+     */
+
+    public static PendingIntent getDeliveryPendingIntent(Context context, String instanceId, int messageId) {
+        Intent sendIntent = new Intent(SMS_DELIVERY_ACTION);
+        sendIntent.putExtra(SMS_INSTANCE_ID, instanceId);
+        sendIntent.putExtra(SMS_MESSAGE_ID, messageId);
+
+        return PendingIntent.getBroadcast(context, messageId, sendIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+    }
+
+
+    /***
+     * Recreates the sent intent that was passed to SMS Manager to check if it exists.
+     * Usefully in cases such as device restarts.
+     * @param context necessary to create pending intent.
+     * @param instanceId identifies the instance of the form being targeted.
+     * @param messageId identifies the specific messages this intent applies to.
+     * @return true if it exists.
+     */
+    public static boolean checkIfSentIntentExists(Context context, String instanceId, int messageId) {
+        Intent sendIntent = new Intent(SMS_SEND_ACTION);
+        sendIntent.putExtra(SMS_INSTANCE_ID, instanceId);
+        sendIntent.putExtra(SMS_MESSAGE_ID, messageId);
+
+        return PendingIntent.getBroadcast(context, messageId, sendIntent, PendingIntent.FLAG_NO_CREATE) != null;
+    }
+
+    /***
+     * Recreates the delivery intent that was passed to SMS Manager to check if it exists.
+     * Usefully in cases such as device restarts.
+     * @param context necessary to create pending intent.
+     * @param instanceId identifies the instance of the form being targeted.
+     * @param messageId identifies the specific messages this intent applies to.
+     * @return true if it exists.
+     */
+    public static boolean checkIfDeliveryIntentExists(Context context, String instanceId, int messageId) {
```

Comment:
Unused method.

Reply:
Yeah, left it there because I was thinking that delivery reports might come in the near future but I guess it can be removed for now.